jujugui/static/gui/src/app/jujulib/payment.js

```diff
@@ -193,33 +192,174 @@ var module = module;
+          - address {Object} The card address object containing
+            - name {String|Null} The name for the address e.g.
+              "Geoffrey Spinach" or "Tuque LTD"
+            - line1 {String|Null} The first address line
+            - line2 {String|Null} The second address line
+            - county {String|Null} The address county
+            - city {String|Null} The address city
+            - postcode {String|Null} The address post code
+            - countryCode {String|Null} The address country code
+          - brand {String} The card brand name
+          - last4 {String} The last four digits of the card number
+          - month {Int} The card expiry month
+          - name {String} The user provided identifier of the card
+          - cardHolder {String} The name of the card owner
+          - valid {Boolean} Whether the card is valid e.g. the card has not
+            expired
+          - Year {Int} The card expiry year
+    */
+    getPaymentMethods: function(username, callback) {
+      const handler = (error, response) => {
+        if (error !== null) {
+          callback(error, null);
+          return;
+        }
+        callback(null, this._parsePaymentMethods(
+          response['payment-methods'], true));
+      };
+      const url = `${this.url}/u/${username}/payment-methods`;
+      return jujulib._makeRequest(this.bakery, url, 'GET', null, handler);
+    },
+
+    /**
+      Create a new payment method.
+
+      @public createPaymentMethod
+      @param name {String} The user's username.
+      @param token {String} A Stripe token.
+      @param callback {Function} A callback to handle errors or accept the
+        data from the request. Must accept an error message or null as its
+        first parameter and a payment method object as its second (see the
+        getPaymentMethods docstring for the fields it returns).
+    */
+    createPaymentMethod: function(username, token, callback) {
+      const handler = (error, response) => {
+        if (error !== null) {
+          callback(error, null);
+          return;
+        }
+        callback(null, this._parsePaymentMethod(response, true));
+      };
+      const id = this._generatePaymentMethodName();
+      const url = `${this.url}/u/${username}/payment-methods/${id}`;
+      const payload = {
+        token: token,
+        'payment-method-name': id
+      };
+      return jujulib._makeRequest(this.bakery, url, 'PUT', payload, handler);
+    },
+
+    /**
+      Remove a new payment method.
+
+      @public removePaymentMethod
+      @param name {String} The user's username.
+      @param id {String} The payment method id.
+      @param token {String} A Stripe token.
+      @param callback {Function} A callback to handle errors or accept the
+        data from the request. Must accept an error message or null as its
+        first parameter and the response as the second parameter.
```

Comment:
What's in the response? Do we care about that? Could the callback just accept an error or null? If not, let's please describe and parse the response accordingly.